### PR TITLE
[11.0][FIX][mail_restrict_follower_selection] fixes a corner case where you could send an email to a partner that you are not supposed to.

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
cc @etobella

**Comment by JordiBForgeFlow** [comment]
@AdriaGForgeFlow @LoisRForgeFlow

File: mail_restrict_follower_selection/models/mail_thread.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        for key in result:
+            for partner_id, email, reason in result[key]:
+                if partner_id:
+                    domain = self.env[
+                        'mail.wizard.invite'
+                    ]._mail_restrict_follower_selection_get_domain()
+                    partner = self.env['res.partner'].search(
+                        [('id', '=', partner_id)] +
+                        safe_eval(domain)
+                    )
```
Comment (etobella) [inline mail_restrict_follower_selection/models/mail_thread.py:22]: You are computing this domain several times, maybe you should declare it before the first key.

**Review by etobella — CHANGES_REQUESTED** [review]

**Review by etobella — APPROVED** [review]
LGTM :+1:

**Review by jarroyomorales — APPROVED** [review]

**Review by AdriaGForgeFlow — APPROVED** [review]

**Pull request merged**